vowpalwabbit/cb_explore_adf_synthcover.cc

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#include "cb_explore_adf_synthcover.h"
+
+#include "cb_explore_adf_common.h"
+#include "reductions.h"
+#include "cb_adf.h"
+#include "rand48.h"
+#include "bs.h"
+#include "gen_cs_example.h"
+#include "cb_explore.h"
+#include "explore.h"
+#include "vw_versions.h"
+#include "version.h"
+#include <vector>
+#include <algorithm>
+#include <cmath>
+
+// All exploration algorithms return a vector of id, probability tuples, sorted in order of scores. The probabilities
+// are the probability with which each action should be replaced to the top of the list.
+
+namespace VW
+{
+namespace cb_explore_adf
+{
+namespace synthcover
+{
+struct cb_explore_adf_synthcover
+{
+private:
+  float _epsilon;
+  float _psi;
+  size_t _synthcoversize;
+  std::shared_ptr<rand_state> _random_state;
+
+  VW::version_struct _model_file_version;
+
+  v_array<ACTION_SCORE::action_score> _action_probs;
```

Comment:
This needs to be initialized in the constructor.

Reply:
Done.